.github/actions/archive-artifacts/action.yaml

```diff
@@ -42,3 +42,22 @@ runs:
             sh -c "kubectl exec -i -n ${NAMESPACE} kcat -- \
                        kcat -L -b ${KAFKA_SERVICE} -t {}  -C -o beginning -e -q -J \
                        > /tmp/artifacts/data/${STAGE}/kafka-messages-{}.log"
+    - name: Dump MongoDB
+      shell: bash
+      continue-on-error: true
+      run: |-
+        set -exu
+
+        ZENKO_MONGODB_DATABASE="${ZENKO_MONGODB_DATABASE:-zenko-database}"
+        MONGODB_ROOT_USERNAME="${MONGODB_ROOT_USERNAME:-root}"
+        MONGODB_ROOT_PASSWORD="${MONGODB_ROOT_PASSWORD:-rootpass}"
+        NAMESPACE="${NAMESPACE:-default}"
+
+        MONGODB_COLLECTIONS=$(kubectl exec -n ${NAMESPACE} data-db-mongodb-sharded-mongos-0 -- mongo ${ZENKO_MONGODB_DATABASE} -u ${MONGODB_ROOT_USERNAME} -p ${MONGODB_ROOT_PASSWORD} --authenticationDatabase admin --quiet --eval "db.getCollectionNames().join(' ')")
+
+        for collection in ${MONGODB_COLLECTIONS}; do
+            kubectl exec -n ${NAMESPACE} data-db-mongodb-sharded-mongos-0 -- mongo ${ZENKO_MONGODB_DATABASE} -u ${MONGODB_ROOT_USERNAME} -p ${MONGODB_ROOT_PASSWORD} --authenticationDatabase admin --quiet --eval "db.getCollection((\"${collection}\")).find().toArray()" > /tmp/artifacts/data/${STAGE}/mongodb-${collection}.log
```

Comment:
can't we use `mongodump` instead?
it allows dumping the whole DB or specific collection, may be faster. However, it exports a BSON: more compact, but maybe less practical to use (for analysis, may need to pass it through `bsondump` or restore it into another mongo)

otherwise, we may use `mongoexport` to dump a collection to JSON, one at a time...

Reply:
Updated with mongodump & bsondump, let's see if my approach is faster...

Reply:
New logic takes 10s to complete, I think we are good...